### PR TITLE
Registration Campaign GA Docs

### PR DESCRIPTION
**Pull request by mjsantani**

### CODE REVIEW

File: api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ms.localizationpriority: medium
+ms.prod: "identity-and-sign-in"
+doc_type: resourcePageType
+---
+
+# authenticationMethodsRegistrationCampaign resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md:14]: GLOBAL: (Please apply this change to all resource topics) 

- Please, remove the beta disclaimer as this is a v1.0 article.

File: changelog/Microsoft.DirectoryServices.json [hunk 2]
```diff
@@ -9235,6 +9235,64 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [registrationEnforcement](https://docs.microsoft.com/en-us/graph/api/resources/registrationEnforcement?view=graph-rest-1.0) resource type.",
+          "Target": "registrationEnforcement"
+        },
+        {
+          "Id": "2df9c094-b6d7-4765-8161-a85fb84c37b9",
+          "ApiChange": "Property",
+          "ChangedApiName": "registrationEnforcement",
+          "ChangeType": "Addition",
+          "Description": "Added the **registrationEnforcement** property to [authenticationMethodsPolicy](https://docs.microsoft.com/en-us/graph/api/resources/authenticationMethodsPolicy?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.DirectoryServices.json:9286]: Please, add "the" before "authenticationMethodsPolicy". Suggested revision:

- Added the \*\*registrationEnforcement\*\* property to the \[authenticationMethodsPolicy\]...

File: api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    {
+      "@odata.type": "microsoft.graph.excludeTarget"
+    }
+  ],
+  "includeTargets": [
+    {
+      "@odata.type": "microsoft.graph.authenticationMethodsRegistrationCampaignIncludeTarget"
+    }
+  ],
+  "snoozeDurationInDays": "Integer",
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md:49]: For consistency, please replace "Integer" with "Int32".

File: api-reference/v1.0/resources/registrationenforcement.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+doc_type: resourcePageType
+---
+
+# registrationEnforcement resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Enforce registration at sign-in time. This can currently only be used to remind users to set up targeted authentication methods (Microsoft Authenticator) using the 'authenticationMethodsRegistrationCampaign`.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/registrationenforcement.md:16]: Please, fix the backtick used to highlight "authenticationMethodsRegistrationCampaign". Suggested revision:

- using the \`authenticationMethodsRegistrationCampaign\`.

Comment (JarbasHorst) [inline api-reference/v1.0/resources/registrationenforcement.md:16]: For clarity, please add "for example," before Microsoft Authenticator. Suggested revision:

- ...(for example, Microsoft Authenticator)...

File: api-reference/v1.0/resources/registrationenforcement.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|authenticationMethodsRegistrationCampaign|[authenticationMethodsRegistrationCampaign](../resources/authenticationmethodsregistrationcampaign.md)|Run campaigns to remind users to setup targeted authentication methods.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/registrationenforcement.md:21]: Please, replace "setup" (noun) with "set up" (verb).

File: api-reference/v1.0/resources/excludetarget.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the users or groups of users that are excluded from a policy.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|id|String|The object identifier of an Azure AD user or group.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/excludetarget.md:21]: Before using acronyms, I recommend you to spell out the words first followed by the acronym in parentheses. However, in this case, I recommend you to just spell out Azure AD. Suggested revision:

- The object identifier of an Azure Active Directory user or group.

File: api-reference/v1.0/resources/authenticationmethodsregistrationcampaignincludetarget.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the users and groups that are targeted for authentication method registration campaigns. Only users and groups that are enabled by the policy to set up the authentication method are targeted.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|id|String|The object identifier of an Azure AD user or group.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaignincludetarget.md:21]: Before using acronyms, I recommend you to spell out the words first followed by the acronym in parentheses. However, in this case, I recommend you to just spell out Azure AD. Suggested revision:

- The object identifier of an Azure Active Directory user or group.

File: api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the settings used to run campaigns to push users to set up targeted authentication methods. Users are prompted to set up the authentication method after they successfully complete a MFA challenge. Only available for the Microsoft Authenticator app for MFA.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|excludeTargets|[excludeTarget](../resources/excludetarget.md) collection|Users and groups of users that are excluded from being prompted to set up the authentication method.|
+|includeTargets|[authenticationMethodsRegistrationCampaignIncludeTarget](../resources/authenticationmethodsregistrationcampaignincludetarget.md) collection|Users and groups of users that are prompted to set up the authentication method.|
+|snoozeDurationInDays|Int32|Specifies the number of days that the user sees a prompt again if they select "Not now" and snoozes the prompt. Minimum 0 days. Maximum: 14 days. If the value is “0” – The user is prompted during every MFA attempt.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md:23]: For consistency, please add colon after Minimum. Suggested revision:

- Minimum: 0 days...

Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md:23]: For consistency, please replace these quotation marks “” with "". Also, please remove the dash and combine the sentences with a comma. Suggested revision:

- If the value is "0", the user is prompted during every MFA attempt.

File: api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+|state|advancedConfigState|Enable or disable the feature. Possible values are: `default`, `enabled`, `disabled`, `unknownFutureValue`. The `default` value is used when the configuration hasn't been explicitly set and uses the default behavior of Azure AD for the setting. The default value is `disabled`.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodsregistrationcampaign.md:24]: For clarity, I recommend you to spell out Azure AD. Suggested revision:

- ...the default behavior of Azure Active Directory for the setting...

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

**Comment by mjsantani** [comment]
@JarbasHorst thank you for the incredibly detailed review. I've addressed all comments - please do take a look.

File: api-reference/v1.0/resources/authenticationmethodspolicies-overview.md [hunk 10]
```diff
@@ -26,6 +27,11 @@ The authentication method policies APIs are used to manage policy settings. For
 |[microsoftauthenticatorauthenticationmethodconfiguration](microsoftauthenticatorauthenticationmethodconfiguration.md)|Define users who can use Microsoft Authenticator on the Azure AD tenant.|
 |[emailauthenticationmethodconfiguration](emailauthenticationmethodconfiguration.md)|Define users who can use email OTP on the Azure AD tenant.|
 
+## Policies available for authentication methods Registration Campaign:
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodspolicies-overview.md:30]: "Registration Campaign" doesn't need to be capitalized. Please replace "Registration Campaign" with "registration campaign".

File: api-reference/v1.0/resources/authenticationmethodspolicies-overview.md [hunk 11]
```diff
@@ -26,6 +27,11 @@ The authentication method policies APIs are used to manage policy settings. For
...
+|Policy       | Description |
+|:---------------------------|:------------|
+|[authenticationMethodsRegistrationCampaign](authenticationmethodsregistrationcampaign.md)| Define users who should be reminded to set up an authentication method (only supported for the Microsoft Authenticator).|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/authenticationmethodspolicies-overview.md:33]: If the information in the parentheses applies to the authentication method, no changes are required. Now, if "only supported for the Microsoft Authenticator" applies to the policy "authenticationMethodsRegistrationCampaign", I recommend you to remove this information from the parentheses and create a new sentence. Suggested revision:

- ...to set up an authentication method. Only supported for the Microsoft Authenticator.

**Review by JarbasHorst — CHANGES_REQUESTED** [review]

File: api-reference/v1.0/resources/authenticationmethodspolicies-overview.md (hunk 11 above)
Comment (mjsantani) [reply api-reference/v1.0/resources/authenticationmethodspolicies-overview.md:33]: Made an update and added "Currently".

**Comment by mjsantani** [comment]
@JarbasHorst - Addressed feedback. Please take a look :)

File: api-reference/v1.0/resources/authenticationmethodspolicies-overview.md (hunk 11 above)
Comment (JarbasHorst) [reply api-reference/v1.0/resources/authenticationmethodspolicies-overview.md:33]: Hi @mjsantani, I don't see this change. Can you please check if it was committed correctly?

**Comment by mjsantani** [comment]
@JarbasHorst - Can we get this approved since all changes have been addressed? I need these to be merged ASAP since the feature is going to be GAing early next week.

**Comment by Lauragra** [comment]
> @JarbasHorst - Can we get this approved since all changes have been addressed? I need these to be merged ASAP since the feature is going to be GAing early next week.

@mjsantani , I see one remaining comment from Jarbas that might not be resolved - please double check? When your feature is GA, please set the ready to merge label on this PR and we'll merge and publish. We don't publish the docs before the feature is available.

**Comment by mjsantani** [comment]
@Lauragra thanks for catching that! I've since fixed the PR and will set the right labels. To confirm, we don't need the second approval before merging?

**Comment by mjsantani** [comment]
@Lauragra @JarbasHorst the schema is live, I've addressed the comments, and changed the labels to "schema live" and "ready to merge". 

Let me know if you need anything else from me to wrap this one up. Thanks!

**Review by Lauragra — APPROVED** [review]

**Pull request merged**